src/java/fr/paris/lutece/plugins/workflow/modules/appointment/web/UpdateAppointmentCancelActionTaskComponent.java

```diff
@@ -106,7 +106,7 @@ public String getDisplayConfigForm( HttpServletRequest request, Locale locale, I
 
         List<Action> listActions = _actionService.getListActionByFilter( filter );
 
-        if ( action.getStateAfter( ).getId( ) == action.getStateBefore( ).getId( ) )
+        if ( action.getStateAfter( ).getId( ) == action.getListIdStateBefore().get(0) )
```

Comment:
Why is the comparison made only on the first state in the list?

Reply:
The revised code now properly checks if `action.getStateAfter().getId()` is present in any of the states in the `listIdStateBefore`. This modification ensures that all possible states of departure are considered
Thank you for your feedback